trino/auth.py

```diff
@@ -421,10 +420,21 @@ def _attempt_oauth(self, response: Response, **kwargs: Any) -> None:
         if not _OAuth2TokenBearer._BEARER_PREFIX.search(auth_info):
             raise exceptions.TrinoAuthError(f"Error: header info didn't match {auth_info}")
 
-        auth_info_headers = parse_dict_header(
-            _OAuth2TokenBearer._BEARER_PREFIX.sub("", auth_info, count=1))  # type: ignore
+        split_challenge = auth_info.split(" ", 1)
```

Comment:
a comment here with an example header value would help. Consider extracting a method/class for this and unit-testing that against some headers you've seen while local testing (and the one mentioned in the issue).

Reply:
The example from the other PR is (comma to join each header value):

```
www-authenticate: Basic realm="Trino", Bearer realm="Trino", token_type="JWT", Bearer x_redirect_server="https://trino.ourdomain.com/oauth2/token/initiate/9e7e5766b1603b9151aa6712f995915658b990c18a6f9a7b350de889651f1ad6", x_token_server="https://trino.ourdomain.com/oauth2/token/c3ade6f9-2a19-1e26-a4a7-0102dbbe8022"
```

So I see this logic would set:

```
split_challenge = ['Basic', 'realm="Trino", Bearer realm="Trino", token_type="JWT", Bearer x_redirect_server="https://trino.ourdomain.com/oauth2/token/initiate/9e7e5766b1603b9151aa6712f995915658b990c18a6f9a7b350de889651f1ad6", x_token_server="https://trino.ourdomain.com/oauth2/token/c3ade6f9-2a19-1e26-a4a7-0102dbbe8022"']
self.scheme = 'Basic'
trimmed_challenge = 'realm="Trino", Bearer realm="Trino", token_type="JWT", Bearer x_redirect_server="https://trino.ourdomain.com/oauth2/token/initiate/9e7e5766b1603b9151aa6712f995915658b990c18a6f9a7b350de889651f1ad6", x_token_server="https://trino.ourdomain.com/oauth2/token/c3ade6f9-2a19-1e26-a4a7-0102dbbe8022"'
```

This confuses me because scheme is actually 3 values here - `Basic`, `Bearer` and `Bearer` but we only pick the first one.

Reply:
Added an example header. Extracted logic to a method and added some testing. Scheme is not used, so I dropped it from the code.